src/main/java/com/reshiftsecurity/plugins/intellij/service/AnalyticsService.java

```diff
@@ -0,0 +1,195 @@
+import java.security.MessageDigest;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AnalyticsService {
+    List<AnalyticsEntry> entries;
+    // NOTE: use https://www.google-analytics.com/debug/collect for debugging
+    private final String ANALYTICS_BASE_URL = "https://www.google-analytics.com/collect";
+    private final String APP_ID = "com.reshiftsecurity.plugins.intellij";
+    private final String APP_ID_KEY = "aid";
+    private final String APP_NAME = "Reshift Intellij Plugin";
+    private final String APP_NAME_KEY = "an";
+    private final String APP_VERSION_KEY = "av";
+    private final String PROTOCOL_VERSION = "1";
+    private final String PROTOCOL_VERSION_KEY = "v";
+    private final String HIT_TYPE = "event";
+    private final String HIT_TYPE_KEY = "t";
+    private final String EVENT_ACTION_KEY = "ea";
+    private final String USER_ID_KEY = "cid";
+    private final String MEASUREMENT_ID_KEY = "tid";
+    private final String ACTION_VALUE_KEY = "cm1";
+    private final String EVENT_VALUE_KEY = "ev";
+    private final String ACTION_CATEGORY_KEY = "ec";
+    private final String ACTION_LABEL_KEY = "el";
+    private final String DOC_PATH = "%2Fintellij";
+    private final String DOC_PATH_KEY = "dp";
+    private final String USER_AGENT = "Reshift Intellij IDE Plugin";
+
+    private String userID;
+    private String applicationVersion;
+    private String measurementID;
+    private String sourceKeywords;
+
+    public AnalyticsService() {
+        this.entries = new ArrayList<>();
+        this.applicationVersion = VersionManager.getVersion();
+        this.userID = getUserIdentifier();
+        this.measurementID = "UA-149586212-2";
+        this.sourceKeywords = System.getProperty("os.name") + " " + USER_AGENT + " " + VersionManager.getVersion();
+    }
+
+    public static AnalyticsService getInstance() {
+        return ServiceManager.getService(AnalyticsService.class);
+    }
+
+    public void recordConsentAction(boolean consent) {
+        AnalyticsAction action = consent ? AnalyticsAction.SETTINGS_GATHER_DATA_YES : AnalyticsAction.SETTINGS_GATHER_DATA_NO;
+        this.entries.add(new AnalyticsEntry(action));
+        this.processActions(true);
+    }
+
+    public void recordAction(AnalyticsAction action, String label) {
+        this.entries.add(new AnalyticsEntry(action, null, label));
+        this.processActions();
+    }
+
+    public void recordAction(AnalyticsAction action) {
+        this.entries.add(new AnalyticsEntry(action));
+        this.processActions();
+    }
+
+    public void recordMetric(AnalyticsAction action, Integer value) {
+        this.entries.add(new AnalyticsEntry(action, value));
+        this.processActions();
+    }
+
+    private static String bytesToHex(byte[] hash) {
+        StringBuffer hexString = new StringBuffer();
+        for (int i = 0; i < hash.length; i++) {
+            String hex = Integer.toHexString(0xff & hash[i]);
```

Comment:
just as in improvement on readability these three lines can be
`result.append(String.format("%02x", aByte));`